src/common/Utility.js

```diff
@@ -499,6 +499,46 @@
             });
         },
         isArray: isArray,
-        template: template
+        template: template,
+        parseVersionString: function(versionString){
+            var _sp = versionString.split('.');
+            return {
+                major: parseInt(_sp[0].replace('v','')),
+                minor: parseInt(_sp[1]),
+                patch: parseInt(_sp[2].split('-')[0]),
+            };
+        },
+        retrofit_114_serialization: function(state,replacement_version) {
+            replacement_version = !replacement_version || replacement_version === '1.14.2-dev' ? '1.18.0' : replacement_version;
+            if(!state.__version)return state;
+            var state_version_obj = this.parseVersionString(state.__version);
+            var target_version_obj = this.parseVersionString(replacement_version);
+            if(state_version_obj.major === 1 && state_version_obj.minor === 14){
+                console.log("Upgrading old persist from "+state.__version+" to "+replacement_version);
+                var _json_str = JSON.stringify(state);
+                _json_str = _json_str.split('"'+state.__version).join('"'+replacement_version);
+    
+                var ret_obj = JSON.parse(_json_str);
```

Comment:
Not sure I would have done it that way, but rather walked (visited) the structure and changed the version per item (and then testing for "graph_Graph" at the same time.

You could also abort early at the first instance of a version > 1.14

Reply:
I considered this but the string split approach seemed less error prone